### PR TITLE
Goodbye Amlogic 3.14

### PR DESCRIPTION
**Pull request by kszaq**
**Motivation and context:**
- LibreELEC focus is on bringing mainline vendor-hack-free support for Amlogic devices (among others).
- Nobody in LE team have time/knowledge to support legacy 3.14 kernel and vendor-specific approach, let alone improving it.
- Users of 3.14 kernel are not left alone as they can use CoreELEC where support is promised until mainline matures.

### CODE REVIEW

File: scripts/image [hunk 1]
```diff
@@ -273,7 +273,7 @@ if [ "$1" = "release" -o "$1" = "mkimage" -o "$1" = "amlpkg" -o "$1" = "noobs" ]
   )
 
   # Create image files if requested
-  if [[ ( "$1" = "amlpkg" || "$1" = "noobs" || "$1" = "mkimage" ) && -n "$BOOTLOADER" ]]; then
+  if [[ ( "$1" = "noobs" || "$1" = "mkimage" ) && -n "$BOOTLOADER" ]]; then
     # INSTALL_SRC_DIR can be board specific
     if [ -n "$DEVICE" -a -d "$PROJECT_DIR/$PROJECT/devices/$DEVICE/install" ]; then
       INSTALL_SRC_DIR="$PROJECT_DIR/$PROJECT/devices/$DEVICE/install"
```
Comment (Kwiboo) [inline scripts/image]: Looks like you can also remove this block that sets `INSTALL_SRC_DIR` (and the passing of it to `scripts/mkimage`), from what I can tell it is only used by amlpkg.

Comment (kszaq) [reply scripts/image]: Thank you, `amlpkg` drop commit updated.

**Comment by kszaq** [comment]
1. Rebased on master.
2. Additional cleanups suggested by @Kwiboo (thank you!).

**Comment by kszaq** [comment]
@raybuntu @adamg88 If you want to introduce 3.14 support back to CE, there is no need to reintroduce BUILD_ANDROID_BOOTIMG - you can boot `uImage` with `bootm` just fine, see https://github.com/kszaq/LibreELEC.tv/blob/amlogic/projects/Amlogic/options#L30-37

File: packages/devel/libcec/package.mk [hunk 2]
```diff
@@ -22,16 +22,6 @@ if [ "$KODIPLAYER_DRIVER" = "bcm2835-driver" ]; then
...
 fi
 
-if [ "$KODIPLAYER_DRIVER" = "libamcodec" ]; then
-  if [ "$TARGET_KERNEL_ARCH" = "arm64" ]; then
-    PKG_CMAKE_OPTS_TARGET="$PKG_CMAKE_OPTS_TARGET -DHAVE_AOCEC_API=1"
-  else
-    PKG_CMAKE_OPTS_TARGET="$PKG_CMAKE_OPTS_TARGET -DHAVE_AMLOGIC_API=1"
-  fi
-else
-  PKG_CMAKE_OPTS_TARGET="$PKG_CMAKE_OPTS_TARGET -DHAVE_AOCEC_API=0 -DHAVE_AMLOGIC_API=0"
```
Comment (jernejsk) [inline packages/devel/libcec/package.mk]: This line should be kept, right?

File: packages/linux/package.mk [hunk 3]
```diff
@@ -65,16 +57,8 @@ if [[ "$KERNEL_TARGET" = uImage* ]]; then
   PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET u-boot-tools:host"
 fi
 
-if [ "$BUILD_ANDROID_BOOTIMG" = "yes" ]; then
-  PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET mkbootimg:host"
-fi
-
 post_patch() {
   cp $PKG_KERNEL_CFG_FILE $PKG_BUILD/.config
-  if [ ! "$BUILD_ANDROID_BOOTIMG" = "yes" ]; then
```
Comment (jernejsk) [inline packages/linux/package.mk]: Content of this IF should be kept.

File: packages/devel/libcec/package.mk (hunk 2 above)
Comment (rayelec) [reply packages/devel/libcec/package.mk]: Isn't that the default?

File: packages/linux/package.mk (hunk 3 above)
Comment (MilhouseVH) [reply packages/linux/package.mk]: Yes... :)

File: packages/devel/libcec/package.mk (hunk 2 above)
Comment (jernejsk) [reply packages/devel/libcec/package.mk]: Probably and it this case I think it doesn't really matter. But usually it's better to be explicit.

**Pull request merged**